fsutil.py

```diff
@@ -15,34 +23,46 @@
 
 # https://stackoverflow.com/questions/8319264/how-can-i-get-the-name-of-a-drive-in-python
 
-def find_circuit_python_user_mode_root():
-    drives = win32api.GetLogicalDriveStrings()
-    logging.debug("raw drives is %s" % drives)
-    drives = drives.split('\000')[:-1]
-    logging.debug("final list of drives %s" % drives)
+def find_circuit_python_user_mode_root(win=use_windows_api):
+    if win:
+        drives = win32api.GetLogicalDriveStrings()
+        logging.debug("raw drives is %s" % drives)
+        drives = drives.split('\000')[:-1]
+        logging.debug("final list of drives %s" % drives)
 
-    # See if one has a volume name we like
-    for drive in drives:
-        volume_name = win32api.GetVolumeInformation(drive)[0]
-        logging.debug("%s volume name is %s" % (drive, volume_name))
-        if volume_name in expected_user_mode_volume_names:
-            return drive
+        # See if one has a volume name we like
+        for drive in drives:
+            volume_name = win32api.GetVolumeInformation(drive)[0]
+            logging.debug("%s volume name is %s" % (drive, volume_name))
+            if volume_name in expected_user_mode_volume_names:
+                return drive
+    else:
+        for drive in os.listdir("/media/{}".format(getpass.getuser())):
+            logging.debug("volume name is %s" % drive)
+            if drive in expected_user_mode_volume_names:
+                return "/media/{}/{}".format(getpass.getuser(), drive)
 
     logging.debug("did not find a suitable drive")
     return None
 
-def find_circuit_python_bootloader_mode_root():
-    drives = win32api.GetLogicalDriveStrings()
-    logging.debug("raw drives is %s" % drives)
-    drives = drives.split('\000')[:-1]
-    logging.debug("final list of drives %s" % drives)
-
-    # See if one has a volume name we like
-    for drive in drives:
-        volume_name = win32api.GetVolumeInformation(drive)[0]
-        logging.debug("%s volume name is %s" % (drive, volume_name))
-        if volume_name in expected_bootloader_mode_volume_names:
-            return drive
+def find_circuit_python_bootloader_mode_root(win=use_windows_api):
+    if win:
+        drives = win32api.GetLogicalDriveStrings()
+        logging.debug("raw drives is %s" % drives)
+        drives = drives.split('\000')[:-1]
+        logging.debug("final list of drives %s" % drives)
+
+        # See if one has a volume name we like
+        for drive in drives:
+            volume_name = win32api.GetVolumeInformation(drive)[0]
+            logging.debug("%s volume name is %s" % (drive, volume_name))
+            if volume_name in expected_bootloader_mode_volume_names:
+                return drive
+    else:
+        for drive in os.listdir("/media/{}".format(getpass.getuser())):
+            logging.debug("volume name is %s" % drive)
+            if drive in expected_bootloader_mode_volume_names:
+                return "/media/{}/{}".format(getpass.getuser(), drive)
```

Comment:
What's the conventional wisdom of `format` vs. `%` these days? You can see I've used `%` for the most part

Reply:
I am not sure generally speaking. But within Circuitpython libraries and projects I tend to see `format()` so I started auto-piloting to that. Though truthfully ever since discovering fstrings (like `f"hello {name_var}"`) that has been more of my goto, but they don't compile in mpy I think so I don't use in circuitpython code.

I don't have a true preference though. Happy with whichever way you like.